pycbc/workflow/core.py

```diff
@@ -764,6 +764,139 @@ def _filename(self, ifo, description, extension, segment):
+                 store_file=True, use_tmp_subdirs=False):
+        """
+        Create a File instance
+        
+        Parameters
+        ----------
+        ifos : string or list
+            The ifo(s) that the File is valid for. If the file is
+            independently valid for multiple ifos it can be provided as a list.
+            Ie. ['H1',L1','V1'], if the file is only valid for the combination
+            of ifos (for e.g. ligolw_thinca output) then this can be supplied
+            as, for e.g. "H1L1V1".
+        exe_name: string
+            A short description of the executable description, tagging
+            only the program that ran this job.
+        segs : glue.segment or glue.segmentlist
+            The time span that the OutFile is valid for. Note that this is
+            *not* the same as the data that the job that made the file reads in.
+            Lalapps_inspiral jobs do not analyse the first an last 72s of the
+            data that is read, and are therefore not valid at those times. If
+            the time is not continuous a segmentlist can be supplied.
+        lfn : string
+            The Logical File Name used to identify this file in the workflow
+        directory : string (optional, default=None)
+            If this *and* store_file=True then the storage path will be set
+            to directory + lfn.
+        tags : list of strings (optional, default=None)
+            This is a list of descriptors describing what this file is. For
+            e.g. this might be ["BNSINJECTIONS" ,"LOWMASS","CAT_2_VETO"].
+            These are used in file naming.
+        """
+        self.metadata = {}
+        
+        # Set the science metadata on the file
+        if isinstance(ifos, (str, unicode)):
+            self.ifo_list = [ifos]
+        else:
+            self.ifo_list = ifos
+        self.ifo_string = ''.join(self.ifo_list)
+        self.description = exe_name
+        
+        if isinstance(segs, (segments.segment)):
+            self.segment_list = segments.segmentlist([segs])
+        elif isinstance(segs, (segments.segmentlist)):
+            self.segment_list = segs
+        else:
+            err = "segs input must be either glue.segments.segment or "
+            err += "segments.segmentlist. Got %s." %(str(type(segs)),)
+            raise ValueError(err)
+            
+        self.tags = tags 
+        if tags is not None:
+            self.tag_str = '_'.join(tags)
+            tagged_description = '_'.join([self.description] + tags)
+        else:
+            tagged_description = self.description
+            
+        # Follow the capitals-for-naming convention
+        self.ifo_string = self.ifo_string.upper()
+        self.tagged_description = tagged_description.upper()
+
+        path = None
+        pfn = None
+
+        if lfn is not None:
+            lfn_path = urlparse.urlparse(lfn).path
+
+            # If it's a file
+            if lfn == lfn_path:
```

Comment:
I wasn't certain about that, but in any case it's consistent with everything else in the dax.